File: Contains Duplicate II/Contains_Duplicate_II.py
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        mpp = {}
        n = len(nums)
        for i in range(n):
            if nums[i] in mpp and abs(mpp[nums[i]] - i) <= k:
                return True
            mpp[nums[i]] = i
        return False
        

File: Contains Duplicate II/Contains_Duplicate_II.java
class Solution {
    public boolean containsNearbyDuplicate(int[] nums, int k) {
        HashMap<Integer, Integer> mpp = new HashMap<>();
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            if (mpp.containsKey(nums[i]) && Math.abs(mpp.get(nums[i]) - i) <= k) return true;
            mpp.put(nums[i], i);
        }
        return false;
    }
}